weblab/entities/views.py

```diff
@@ -91,32 +107,72 @@ def get_success_url(self):
 
 
 class ModelEntityListView(LoginRequiredMixin, ModelEntityTypeMixin, ListView):
+    """
+    List all user's model entities
+    """
     template_name = 'entities/entity_list.html'
 
     def get_queryset(self):
         return self.model.objects.filter(author=self.request.user)
 
 
 class ProtocolEntityListView(LoginRequiredMixin, ProtocolEntityTypeMixin, ListView):
+    """
+    List all user's protocol entities
+    """
     template_name = 'entities/entity_list.html'
 
     def get_queryset(self):
         return self.model.objects.filter(author=self.request.user)
 
 
-class ModelEntityVersionView(LoginRequiredMixin, ModelEntityTypeMixin, VersionMixin, DetailView):
+class EntityAccessMixin(UserPassesTestMixin):
```

Comment:
I like this mix-in approach! But not sure about the "access" part of the name, as the checks concern _visibility_. Maybe use something like EntityViewingMixin or EntityVisibleMixin to emphasize it doesn't imply editing rights?

Reply:
That might be better, thank you.

Reply:
I like `EntityVisibleMixin`.